.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,15 @@
+name: ShellCheck
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: shellcheck ./scripts/*
```

Comment:
:100: top! we should have added shellcheck already!

Could we add it as a `make shellcheck` target, with a gha job that calls it, and also have it run as part of `make test`?

Reply:
indeed!

Reply:
done!